New article to solutions library - Media, text to audio conversion (#274)
mongodb/docs-atlas-architecture: source/solutions-library/text-to-audio-conversion.txt

**Partners:** `Google NotebookLM <https://notebooklm.google/>`__

Solution Overview
-----------------

The surge in demand for audio content has prompted news organizations to
seek efficient ways to deliver daily summaries. For example, podcasts
have `9 million listeners per year
<https://advertise.acast.com/news-and-insights/future-of-podcast-advertising-in-2025-industry-leaders-predictions>`__
in the U.S. alone. However, automating this process is challenging
because it involves managing dynamic article data and converting it into
high-quality audio experiences.

With MongoDB and generative AI, you can build a news automation solution
to streamline and scale podcast creation. MongoDB serves as the core
data layer for the system, efficiently managing news articles as
flexible, schema-less documents within a single collection. These
documents capture both static information—such as title, content, and
publication date—and dynamic metrics that monitor article performance
and popularity over time, such as the number of qualified reads. You can
also enrich derived insights, such as sentiment analysis and key
entities, with a generative AI pipeline and store them in the same

> MongoDB collection. 

Author: Ok

   articles from the database.

#. **Generate Podcast Script:** Pass the articles through an AI pipeline to
   create a structured, multi-voice podcast script.

#. **Convert to Audio:** Use advanced text-to-speech models to transform the
   script into high-quality audio, stored as a ``.wav`` file.

#. **Optimize Delivery:** Cache the generated podcast to ensure seamless,
   on-demand playback for users.

This framework delivers high-quality, human-like narration in MP3
format, providing users with a professional and engaging listening
experience.

Build the Solution
------------------

Follow these steps to build a text-to-audio solution using the
MongoDB `ist.media <https://github.com/mongodb-industry-solutions/ist.media/tree/main>`__
GitHub repository. You can use this framework as inspiration to 
build your own customized text-to-audio pipeline.

.. procedure::
   :style: normal

   .. step:: Deploy the ist.media demo

      Clone the `ist.media github repository
      <https://github.com/mongodb-industry-solutions/ist.media/tree/main>`__
      and follow the ``README`` instructions to deploy the demo. 

   .. step:: Create a feed for news

>       Run the demo and verify that the /feed endpoint provides the news

Author: Ok

      feed for the current day. Alternatively, if you prefer not to use
      the ist.media news collection mechanisms, you can supply your own

>       data—served statically by the endpoint in the same format.

Author: Ok

   .. step:: Generate text-to-audio conversion

      Run the `podcast.py
      <https://github.com/mongodb-industry-solutions/ist.media/blob/main/scripts/podcast.py>`__
      file in the ist.media demo. This script uses the `AutoContent API
      <https://autocontentapi.com/>`__ to generate the podcast. It then
      downloads and saves it with the date (day/month/year) in the
      filename.

Key Learnings
-------------

To create a media solution that converts news data into audio content,
you need a system that is flexible, fast, and able to scale easily.
MongoDB makes this possible through these core strengths:

- **The document model handles diverse attributes:** News data combines
  various attributes, including static fields (e.g., ID, title, body,
  date, image URL), dynamic metadata (e.g., read count), AI-generated
  insights (e.g., keywords, sentiment), and embeddings for semantic
  search. The document model supports all these elements, removing

>   database limitations and allowing the system to evolve smoothly.

Author: To answer your question. That would be great, I think it's easier for the reader to leave them in a clear way the main takeaways.

>   database limitations and allowing the system to evolve smoothly.

Author: Ok

- **Speed ensures operational efficiency:** By processing complete,
  self-contained documents, MongoDB avoids complex operations, enabling
  faster analysis and near real-time transformation of articles into
  audio content.

- **Scalable systems enable growth:** MongoDB Atlas
  scalable systems handle small changes and large amounts of data
  smoothly, ensuring high performance and reliability as your media

>   application grows.

Author: I made a change to only leave MongoDB ATlas handles, to avoid the use of two close s's

      own data, which is served statically by the endpoint in the same
      format.

   .. step:: Generate text-to-audio conversion

      Run the `podcast.py
      <https://github.com/mongodb-industry-solutions/ist.media/blob/main/scripts/podcast.py>`__
      script in the ist.media demo. This script uses the `AutoContent API
      <https://autocontentapi.com/>`__ to generate the podcast. It then
      downloads and saves it with the date (day/month/year) in the
      filename.

Key Learnings
-------------

To create a media solution that converts news data into audio content,
you need a system that is flexible, fast, and able to scale easily.
MongoDB makes this possible through these core strengths:

- **The document model handles diverse attributes:** News data combines
  various attributes, including static fields such as ID, title, body,
  date, and image URL; dynamic metadata such as read count; AI-generated
  insights such as keywords and article sentiment; and embeddings for
  semantic search. The document model supports all these elements,
  removing database limitations and allowing the system to evolve

>   smoothly.

Reviewer: suggestion: avoid using [semicolons](https://www.mongodb.com/docs/meta/style-guide/style/punctuation/#semicolons) - recommend changing to commas and maybe condensing

.. facet::
   :name: genre
   :values: tutorial

.. meta::
   :keywords: media, videos, news, gen AI 
   :description: Automate news broadcasting combining generative AI with MongoDB.
   
.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

Combine generative AI for podcast creation and MongoDB for data storage
to automate and scale news broadcasting.

**Use cases:** `Gen AI
<https://www.mongodb.com/use-cases/artificial-intelligence>`__

**Industries:** Media

**Products:** `MongoDB Atlas <http://mongodb.com/atlas>`__, 
`MongoDB Aggregation Framework
<https://www.mongodb.com/docs/manual/core/aggregation-pipeline/>`__,
`MongoDB Atlas Vector Search
<https://www.mongodb.com/products/platform/atlas-vector-search>`__

**Partners:** `Google NotebookLM <https://notebooklm.google/>`__

Solution Overview
-----------------

The surge in audio content demand like podcasts, with `9 million new
listeners in the U.S. alone
<https://advertise.acast.com/news-and-insights/future-of-podcast-advertising-in-2025-industry-leaders-predictions>`__,
has prompted news organizations to seek efficient ways to deliver daily

> summaries. Automating this process remains challenging, requiring

Author: I like the second option to keep the stats in the article

managing dynamic articles data to convert it into high-quality audio
experiences. 

With MongoDB and generative AI, you can build a news automation solution
to streamline and scale podcast creation. MongoDB serves as the core
data layer for the system, efficiently managing news articles as
flexible, schema-less documents within a single collection. These
documents capture both static information—such as title, content, and
publication date—and dynamic metrics, like the number of qualified

> reads, that monitor article performance and popularity over time.

Author: Ok

Moreover, derived insights, such as sentiment analysis and key entities,
can be enriched through a generative AI pipeline and added to the same

> collection for analysis.

Author: Ok

This adaptable structure ensures efficiency and scalability, providing a
robust framework to query and extract the latest news and metadata. By
integrating advanced language models, this information can be further
transformed into audio podcasts. With this foundation in place, you
unlock new AI-driven business opportunities, attract new customers and
increase revenue streams.

Reference Architectures
-----------------------

To implement this framework, you'll need MongoDB for data storage and
AI-powered speech synthesis for audio creation. You can use Google’s
NotebookLM model to refine news text with accurate intonation and
pacing. The steps below outline the workflow for converting news

> summaries into audio:

Author: Ok

.. figure:: /includes/images/industry-solutions/text-to-audio-architecture.svg
   :figwidth: 1200px
   :alt: visualization for the network chatbot
   
   Figure 1. AI-based text-to-audio conversion architecture

- **Retrieve Articles:** Use the `Aggregation Framework
  <https://www.mongodb.com/docs/manual/aggregation/>`__ and `Vector
  Search
  <https://www.mongodb.com/docs/atlas/atlas-vector-search/vector-search-overview/>`__,
  to fetch relevant news articles from the database.

- **Generate Podcast Script:** Pass the articles through an AI pipeline to
  create a structured, multi-voice podcast script.

- **Convert to Audio:** Use advanced text-to-speech models to transform the
  script into high-quality audio, stored as a .wav file.

- **Optimize Delivery:** Cache the generated podcast to ensure seamless,

>   on-demand playback for users.

Author: Ok

This framework delivers high-quality, human-like narration in MP3
format, providing users with a professional and engaging listening
experience.

Build the Solution
------------------

This solution is part of the larger ist.media project. You can use this
framework as inspiration for building your own customized text-to-audio
solution. To do so, follow these steps:

.. procedure::
   :style: normal

   .. step:: Deploy the ist.media demo

      Clone the `ist.media github repository
      <https://github.com/mongodb-industry-solutions/ist.media/tree/main>`__

>       and follow the Readme instructions to deploy the demo. 

Author: Ok

   .. step:: Create a feed for news

      Run the demo and verify that the /feed endpoint provides the news
      feed for the current day. Alternatively, you can supply your own
      data—served statically by the endpoint in the same format—if you
      prefer to skip the initial news collection mechanisms used by
      ist.media.

   .. step:: Generate text-to-audio conversion

      Run the `podcast.py
      <https://github.com/mongodb-industry-solutions/ist.media/blob/main/scripts/podcast.py>`__
      file in the ist.media demo. This script uses the notebooklm.google

>       API to generate the podcast, downloads it from storage, and saves

Author: it was this one https://autocontentapi.com/ 

Create professional NotebookLM podcasts instantly using our powerful API solution for podcast generation, powered by Google AI technology

>       API to generate the podcast, downloads it from storage, and saves

Author: Ok

      it with the date in the filename (day/month/year).

Key Learnings
-------------

Building a news automation solution that transforms data into engaging
audio content requires systems designed for flexibility, speed,

> scalability, and adaptability. Below are the key insights that make this
> approach successful:

Reviewer: I suggest changing the structure of this sentence for readability and relevance: 

```suggestion
scalability, and adaptability. The following aspects of
MongoDB make this approach successful:
```

Author: Ok

- **The document model handles diverse attributes:** News data combines
  various attributes, including static fields (e.g., ID, title, body,
  date, image URL), dynamic metadata (e.g., read count), AI-generated
  insights (e.g., keywords, sentiment), and embeddings for semantic
  search. The document model supports all these elements, removing

>   database limitations and allowing the system to evolve smoothly.

Author: Not implemented, left with previous language to highlight benefits to media organizations

> - **Speed drives efficiency:** By processing complete, self-contained
>   documents, the system avoids complex operations, enabling faster
>   enrichment, analysis, and transformation of articles into audio
>   content in near real-time.

Reviewer: Suggestion for brevity. Also, can this be connected more to MongoDB's performance?
```suggestion
- **Performance:** By processing self-contained
  documents, the system can quickly transform
  articles into audio content.
```

Author: Adapted to this synthax

- **Speed drives efficiency:** By processing complete, self-contained
  documents, MongoDB avoids complex operations, enabling faster analysis
  and near real-time transformation of articles into audio content.

- **Scalability ensures reliable growth and performance:** Whether managing
  small updates or processing large volumes of dynamic data, a scalable
  system maintains high availability and supports seamless expansion,
  making it ideal for handling the demands of large-scale media

>   applications.

Author: Implemented changes to this

- **Scalability supports growth and performance:** MongoDB scalable systems
  handle small changes and large amounts of data smoothly, ensuring high
  performance and reliability as your media application grows.

- **Flexibility empowers developers:** Without rigid schemas, new data points
  like AI insights, audience metrics, or editorial updates can be
  integrated seamlessly, allowing experimentation, scaling, and

>   adaptation to evolving news consumption.

Author: Adapted to this 

- **Flexible systems empower developers:** Without fixed schemas,
  developers can easily add new information—like AI insights, audience
  metrics, or editorial updates—making it simple to adapt and respond to
  evolving news consumption.
